src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,62 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exeptionforservice.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MINIMUM_PASSWORD_LENGTH = 6;
+    private static final int MINIMUM_LOGIN_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        isUserNull(user);
+        validatedAge(user.getAge());
+        validatePassword(user.getPassword());
+        validateLogin(user.getLogin());
+        checkUserInStorage(user);
+        return storageDao.add(user);
+    }
+
+    private void isUserNull(User user) {
+        if (user == null) {
+            throw new RegistrationException("User can not be null!");
+        }
+    }
```

Comment:
whenever you create method with "is" in the name, we always expect to see false or true as a return. 
So you have two options either rename to validateUser() or return boolean.